lib/site-inspector.rb

```diff
@@ -26,25 +21,86 @@
 class SiteInspector
 
   def self.load_data(name)
+    require 'yaml'
     YAML.load_file File.expand_path "./data/#{name}.yml", File.dirname(__FILE__)
   end
 
-  def initialize(domain)
+  # Utility parser for HSTS headers.
+  # RFC: http://tools.ietf.org/html/rfc6797
+  def self.hsts_parse(header)
```

Comment:
Your call. Not sure if there's more to it / it's useful elsewhere / you want to maintain it. Even a `./lib/hsts_parser.rb` would feel better.